runtime/auth/aws-signing-tests/common/src/aws/smithy/kotlin/runtime/auth/awssigning/tests/MiddlewareSigningTestBase.kt

```diff
@@ -28,7 +28,9 @@ import kotlin.test.assertEquals
 @Suppress("HttpUrlsUsage")
 @OptIn(ExperimentalCoroutinesApi::class)
 public abstract class MiddlewareSigningTestBase : HasSigner {
-    private fun buildOperation(streaming: Boolean = false, replayable: Boolean = true): SdkHttpOperation<Unit, HttpResponse> = SdkHttpOperation.build {
+    private var readFromChannel: SdkByteReadChannel? = null
```

Comment:
**Suggestion**: Storing the read channel as a side-effect of `buildOperation` works but it seems messy. Consider returning `Pair<SdkHttpOperation<Unit, HttpResponse>, SdkByteReadChannel?>` from `buildOperation` instead. Tests which need it can get it as `val (op, channel) = buildOperation(...)`. Tests which don't can either use `val (op, _) = buildOperation(...)` or use a convenience function (e.g., `buildOperation` which calls `buildOperationWithChannel`).